core/lib/zksync_core/src/api_server/web3/namespaces/eth.rs

```diff
@@ -861,9 +861,8 @@ impl<G: L1GasPriceProvider> EthNamespace<G> {
                     .map_err(|err| internal_error(METHOD_NAME, err))?;
                 *from_block = logs
                     .last()
-                    .map(|log| MiniblockNumber(log.block_number.unwrap().as_u32()))
+                    .map(|log| MiniblockNumber(log.block_number.unwrap().as_u32() + 1))
                     .unwrap_or(*from_block);
```

Comment:
AFAIU, you could increase the cursor here as well to at least `*from_block + 1`. More precisely, in both cases (whether `logs` are empty or not), the cursor can be increased to `to_block + 1`. The current implementation will query redundant miniblock ranges for events that we know (by performing previous requests) are not there. 

There's a caveat: there seems to be a TOCTOU possibility with the `*from_block != to_block` check, and generally, fetched events may belong to a greater miniblock that was inserted into the DB after `to_block` was fetched. IMO, it would make sense to refactor `get_logs()` so that it takes a concrete range of miniblock numbers – `from_block..=to_block` – as an input. This would make this logic easier to reason about and would probably simplify DAL logic as well.

Reply:
The thing is that if `from_block` is greater than actual latest block number and the query occurred before adding the block with that number, cursor will move forward anyway and changes from this block won't be returned. If talking about moving it to `to_block + 1` we are actually thinking of doing it that way, but again, if user manually entered `to_block` greater then latest number of block, we should move it to the latest current block number